lfs: add tests for rate limit retries

sisp opened the pull request:
I've added some tests for rate limit retries.

Follow-up of #340.

/cc @shcheklein

shcheklein commented on tests/test_lfs.py:
```diff
@@ -74,3 +85,197 @@ def test_lfs(tmp_dir: TmpDir, scm: Git, lfs_objects: TmpDir):
+        lambda: {"X-RateLimit-Reset": f"{int(time()) + 1}"},
+    ],
+)
+def test_rate_limit_max_retries_batch(
```
it seems this already covered by the test above?

sisp replied:
Do you mean the parametrization with the different rate limit headers? Or what specifically do you think is already covered? 🤔 

This test ensures that rate limit retries for the batch API don't continue indefinitely.

shcheklein replied:
okay, yes, I see the difference 👍

shcheklein commented on tests/test_lfs.py:
```diff
@@ -74,3 +85,197 @@ def test_lfs(tmp_dir: TmpDir, scm: Git, lfs_objects: TmpDir):
+        with pytest.raises(ClientResponseError, match="Too many requests"):
+            client.download(storage, [Pointer(oid=FOO_OID, size=FOO_SIZE)])
+
+        results = recorder[lfs_batch_url]
```
can we simplify the test and use `lfs_batch_url` to test it, instead of `lfs_object_url`?

sisp replied:
I'm not sure I understand your proposal. Here, I'd like to make sure that rate limit retries work for the object API.

shcheklein replied:
yes, I see the difference now

shcheklein commented on tests/test_lfs.py:
```diff
@@ -74,3 +85,197 @@ def test_lfs(tmp_dir: TmpDir, scm: Git, lfs_objects: TmpDir):
+    recorder = CallbackResultRecorder()
+
+    with aioresponses() as m:
+        m.post(
```
ok, now that I see the difference between tests, can we move each of these `m.post`, `m.get` blocks into a separate function and reuse them across tests? so that each test is small in size (and probably will be move descriptive)? wdyt?

sisp replied:
Good idea! :+1: I've pushed a commit with refactored tests. WDYT?

shcheklein reviewed (approved):
thanks @sisp ! good stuff. Feel free to merge (we usually do squash merge I think) and probably you should be able to trigger the new release (releases -> create a new release  I think). We can then bump the version required on the DVC side if you feel it is needed.

sisp commented:
Thanks for the helpful review, @shcheklein! 🙇 

IMO, no release is needed, as the only change since the last one is the added tests by this PR. [v3.3.0](https://github.com/iterative/scmrepo/releases/tag/3.3.0) already includes the feature. No need for a bump in DVC from my side, [`scmrepo>=3,<4`](https://github.com/iterative/dvc/blob/017a510b83a682ddf4059a3d33bb32bc63324bba/pyproject.toml#L63) includes v3.3.0.
